.github/workflows/update-jira-tickets.yaml

```diff
@@ -0,0 +1,120 @@
+          else
+            currentTag="${{ github.event.inputs.tagName }}"
+          fi
+          echo "Tag: ${currentTag}"
+          echo "tag=${currentTag}" >> $GITHUB_OUTPUT
+
+      - name: Validate tag
+        run: |
+          git fetch -q --tags
+          if ! git tag --list | grep -qx "${{ steps.get-tag.outputs.tag }}"; then
+            echo "Not a valid tag, ending workflow"
+            exit 1
+          fi
+
+      - name: Check if tag is a release candidate
+        run: |
+          if [[ ${{ steps.get-tag.outputs.tag }} != *"-rc"* ]]; then
+            echo "Not a release candidate, ending workflow."
+            exit 1
+          fi
+
+      - name: Ping Jenkins for previous successful tag
+        id: ping-jenkins
+        run: |
+          repoName=${GITHUB_REPOSITORY##*/}
+          currentTag="${{ steps.get-tag.outputs.tag }}"
+          git fetch -q --tags
+          rcTags=$(git tag --sort=-creatordate | grep -- -rc)
+          started=false
+          previousTag=""
+
+          for tagName in $rcTags
+          do
+            # Skip tags more recent than the current one
+            if [ "$tagName" != "$currentTag" ] && [ "$started" = false ]; then
+              continue;
+            elif [ "$tagName" == "$currentTag" ]; then
+              started=true;
+              continue;
+            fi
+
+            jenkinsUrl="${JENKINS_HOST}/job/${repoName}/view/tags/job/${tagName}/lastSuccessfulBuild"
+            echo "Checking for successful Jenkins build for GitHub tag ${tagName} at ${jenkinsUrl}"
+            
+            response=$(curl -o /dev/null --silent -w "%{http_code}\n" -u "${JENKINS_USER}:${JENKINS_TOKEN}" "${jenkinsUrl}/api/json")
+            echo "Response: ${response}" 
+            if [ "$response" == "200" ]; then
+              echo "Found successful Jenkins build for GitHub tag ${tagName}: ${jenkinsUrl}"
+              previousTag="${tagName}"
+              break
+            fi
+          done
+          
+          if [ "${previousTag}" == "" ]; then
+            echo "No successful Jenkins builds found for any previous tags. Ending workflow"
+            exit 1
+          else
+            echo "tagName=${previousTag}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Determine Jira tickets from commit messages
+        id: jira-tickets
+        run: |
+          currentTag="${{ steps.get-tag.outputs.tag }}"
+          previousTag="${{ steps.ping-jenkins.outputs.tagName }}"
+          commitMessages=$(git log --pretty=%B $previousTag..$currentTag)
+          echo "Commit messages since the last release: ${commitMessages}"
+          jiraTickets=$(echo "$commitMessages" | grep -io 'M2-[0-9]\+' | tr '[:lower:]' '[:upper:]' | sort | uniq | tr '\n' ' ')
+          echo "Jira tickets since the last release: ${jiraTickets}"
+          echo "tickets=${jiraTickets}" >> $GITHUB_OUTPUT
+
+      - name: Periodically ping Jenkins for current tag build status
```

Comment:
Can this step with a while true be in loop forever? If not at leat will be running for several minutes and can use all our github action build minutes

Can we change this to be trigger by jenkins after build finishs? So when we move from jenkins to GHA, it can me triggered after a GHA build

Reply:
It should only run as long as the Jenkins build runs, so the only way it should run forever is if the Jenkins build never exits. I'd rather not move any of this logic back to Jenkins so I'm open to alternatives

Reply:
Could a max number of retries or a timeout be set to handle the case of a runaway Jenkins build that never exits?

Reply:
It could! One potential issue with the external limit though is that we won't know how long an arbitrary Jenkins build will take, and this workflow may exit prematurely because of it. Given that Jenkins releases to the test environment are triggered manually, there's usually someone keeping an eye on it in order to validate the release afterwards. That makes the runaway scenario unlikely

Reply:
Yes, I was thinking making it some high number that should never be reached on a normal build run (eg. 360 which should be 6 hours). So
- if someone is keeping an eye on it as might be the case most of the time, they can kill the Jenkins build which stops the loop in GH
- if the person steps away and forgets, we still have a max retry safety mechanism that kicks in after x number of hours (eg. 6 hours). Which lose at most 360 minutes of the 2000-3000 monthly minutes the org has (not sure what our [monthly plan](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes) gives us)

Reply:
Sounds reasonable. I'll update it

Reply:
So awesome. Thanks @sultanofcardio